File: 1 - metodosBusca/pacoteBusca_JomiHubner/src/grafo/Vertice.java
package grafo;

import java.util.*;

public class Vertice {
	Set<Aresta> arestas = new HashSet<Aresta>();
	
	private int id;
	
	public Vertice(int id) {
		this.id = id;
	}
	
	public void addAresta(Aresta a) {
		arestas.add(a);	
	}
	
	public int getId() {
		return id;
	}
	
	public List<Vertice> getAdjacentes() {
		List<Vertice> adj = new ArrayList<Vertice>();
		for (Aresta a: arestas) {
			if (a.vi == this) {
				adj.add(a.vj);
			} else {
				adj.add(a.vi);
			}
		}
		return adj;
	}

    public Map<Vertice,Integer> getCustoAdjacentes() {
        Map<Vertice,Integer> adj = new HashMap<Vertice,Integer>();
        for (Aresta a: arestas) {
            ArestaValorada ac = (ArestaValorada)a; // TODO: is it ok?
            if (a.vi == this) {
                adj.put(a.vj, ac.getCusto());
            } else {
                adj.put(a.vi, ac.getCusto());
            }
        }
        return adj;
    }

    public Set<Aresta> getArestas() {
        return arestas;
    }
	
	public String toString() {
		return String.valueOf(id);
	}
}

File: 1 - metodosBusca/python/app/core/grafo/vertice.py
from typing import List, Dict, Set
from .aresta import Aresta

class Vertice:
    def __init__(self, id: int):
        self.id = id
        self.arestas: Set[Aresta] = set()

    def add_aresta(self, a: 'Aresta'):
        self.arestas.add(a)

    def get_id(self) -> int:
        return self.id

    def get_adjacentes(self) -> List['Vertice']:
        adj = []
        for a in self.arestas:
            if a.vi == self:
                adj.append(a.vj)
            else:
                adj.append(a.vi)
        return adj

    def get_custo_adjacentes(self) -> Dict['Vertice', int]:
        adj = {}
        for a in self.arestas:
            ac = a  # Assuming all edges are valued edges
            if a.vi == self:
                adj[a.vj] = ac.get_custo()
            else:
                adj[a.vi] = ac.get_custo()
        return adj

    def get_arestas(self) -> Set[Aresta]:
        return self.arestas

    def __str__(self) -> str:
        return str(self.id)